dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -43,9 +43,432 @@
+            )
+
+        elif nin == 2:
+            if inplace is True:
+                binary_inplace_fn = getattr(ti, _name + "_inplace")
+            else:
+                binary_inplace_fn = None
+
+            def _call_func(src1, src2, dst, sycl_queue, depends=None):
+                """
+                A callback to register in UnaryElementwiseFunc class of
+                dpctl.tensor
+                """
+
+                if depends is None:
+                    depends = []
+
+                if mkl_call is True:
+                    mkl_fn_to_call = getattr(vmi, "_mkl" + _name + "_to_call")
+                    mkl_impl_fn = getattr(vmi, _name)
+                    if mkl_fn_to_call is not None and mkl_fn_to_call(
+                        sycl_queue, src1, src2, dst
+                    ):
+                        # call pybind11 extension for binary function from OneMKL VM
+                        return mkl_impl_fn(sycl_queue, src1, src2, dst, depends)
+                return dpt_impl_fn(src1, src2, dst, sycl_queue, depends)
+
+            self.func = BinaryElementwiseFunc(
+                name,
+                dpt_result_type,
+                _call_func,
+                docs,
+                binary_inplace_fn,
+                acceptance_fn=acceptance_fn,
+            )
+        else:
+            raise NotImplementedError
+
+        self.__doc__ = docs
+        self.__name__ = "ufunc"
+
+    def __call__(
+        self,
+        *args,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+        **kwargs,
+    ):
+        dpnp.check_supported_arrays_type(
+            *args, scalar_type=True, all_scalars=False
+        )
+        if kwargs:
+            raise NotImplementedError(
+                f"Requested function={self.name_} with kwargs={kwargs} "
+                "isn't currently supported."
+            )
+        if where is not True:
+            raise NotImplementedError(
+                f"Requested function={self.name_} with where={where} "
+                "isn't currently supported."
+            )
+        if subok is not True:
+            raise NotImplementedError(
+                f"Requested function={self.name_} with subok={subok} "
+                "isn't currently supported."
+            )
+        if dtype is not None and out is not None:
+            raise TypeError(
+                f"Requested function={self.name_} only takes `out` or `dtype`"
+                "as an argument, but both were provided."
+            )
+        if order is None:
+            order = "K"
+        elif order in "afkcAFKC":
+            order = order.upper()
+        else:
+            raise ValueError(
+                f"order must be one of 'C', 'F', 'A', or 'K' (got '{order}')"
+            )
+
+        if self.nin_ == 1:
+            if out is None and len(args) == 2:
+                out = args[1]
```

Comment:
Could you please describe a bit the use case when input might become `out` parameter? (specially considering that `self.nin_ == 1`)